docker/Dockerfile.php

```diff
@@ -1,40 +1,48 @@
 FROM php:7.1.33-fpm AS base
 WORKDIR /var/www/html
 
-RUN apt update -y
-RUN DEBIAN_FRONTEND=noninteractive apt-get install --no-install-recommends -y \
-    cron \
-    git \
-    libicu-dev \
-    libzip-dev \
-    procps \
-    python3-pip \
-    python3-setuptools \
-    unzip \
-    vim
-RUN apt purge -y --auto-remove
+ENV DEBIAN_FRONTEND=noninteractive
 
-RUN docker-php-ext-install \
+RUN apt-get update -y \
+ && apt-get install --no-install-recommends -y \
+    cron=3.0pl1-134+deb10u1 \
+    git=1:2.20.1-2+deb10u9 \
+    libicu-dev=63.1-6+deb10u3 \
+    libzip-dev=1.5.1-4 \
+    procps=2:3.3.15-2 \
+    python3-pip=18.1-5 \
+    python3-setuptools=40.8.0-1 \
+    unzip=6.0-23+deb10u3 \
+ # TODO(timw): Unclear what the next line should remove, it does
+ # not seem to clear caches which needs to be done manually.
+ && apt-get purge -y --auto-remove \
+ && apt-get clean \
+ && rm -rf /var/lib/apt/lists/* \
+# TODO(timw): Can we be explicit about versions here?
```

Comment:
I believe the versions here are fixed to the version of the docker image you install, in that the source comes with the image. It's not downloaded at creation time.